### ISSUE TITLE
Error reading layer in a GDB on a read-only drive QGIS 3.28.8

### ISSUE THREAD

**Issue by LauraBRegional** [qgis/QGIS#53918]
### What is the bug or the crash?

We are having problems reading data on a read-only drive. With QGIS 3.22 this problem does not exist, but with 3.28.6 and 3.28.8 it does.
When elements are imported into QGIS from a .qlr, they appear with a broken link, as if the layer is not in the given path.
When we enter the layer directly, we are not able to visualize the attribute table but we can see the geometry.

Thank you in advance

### Steps to reproduce the issue

1. Have an only-read space in your drive
2. Add vector layer from .qlr or directly from .gdb
3. From .qlr you have the link error to the file, directly from gbd, you can see the geometry but not the attribute table

### Versions

3.28.6
3.28.8

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by agiudiceandrea** [comment]
@LauraBRegional, thanks for reporting. Could you please provide the content of the QGIS Help -> About menu ("click in the table, Ctrl+A and then Ctrl+C. Finally paste here.") as requested by the issue report form? Could you provide the GPKG layer or a sample GPKG layer with which the issue occurs?

**Comment by LauraBRegional** [comment]
Hi Andrea,

Here you havem y information from the About menu:
QGIS version
3.28.6-Firenze
QGIS code revision
868c9fa03b<https://github.com/qgis/QGIS/commit/868c9fa03b>
Qt version
5.15.3
Python version
3.9.5
GDAL/OGR version
3.6.4
PROJ version
9.2.0
EPSG Registry database version
v10.082 (2023-02-06)
GEOS version
3.11.2-CAPI-1.17.2
SQLite version
3.41.1
PDAL version
2.5.2
PostgreSQL client version
unknown
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.13.1
OS version
Windows 10 Version 2009
Active Python plugins
cartografia_br
1000
OpenICGC
1.1.15
pluginbuilder3
3.2.1
plugin_reloader
0.9.3
qgis2web
3.16.0
QlrBrowser
3.0.0
SentinelHub
2.0.0
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99
sagaprovider
2.12.99


We tried also with 3.28.8 versions but had the same problem. The 3.28 versions have both this issue, but not the 3.22 one.

Related to provide a GPKG layer or a sample GPKG layer, we can send you a file exemple but the issue is not in the geometry or in the layer. Maybe I didn’t explain correctly, sorry for my english.

Our issue is with all of layers that are stored in an only-read drive in the company. We are sure that the files are in the path that provide the .qlr file, but when these layers in a .qlr file have filters, when QGIS add these layers appear with a broken path


Example:
***@***.***

Source
F:/ETRS89/Carto/Base_labels/labels.gdb|layername=tl25m_15000_5001_2|subset=”PLACEMENT” = 0
( And this is correct)


If I try to make a GPKG with the broken paths I gave an error because there is no data....

For all the layers stored in our only-read drive (added with add vector or .qlr files) we can see the geometries but not the attribute table, it appears empty (blank)


Thank you!




                                  Laura Vergoñós
                                  Ambientòloga i analista SIG
                                  +34  93 223 82 98<tel:+34932238298>
[email-V3a-2.png]
P   Imprimeix només si és necessari

De: Andrea Giudiceandrea ***@***.***>
Enviado el: 25 July 2023 11:51
Para: qgis/QGIS ***@***.***>
CC: Laura Vergoñós Pascual ***@***.***>; Mention ***@***.***>
Asunto: Re: [EXTERN] [qgis/QGIS] Error reading layer in a GDB on a read-only drive QGIS 3.28.8 (Issue #53918)


@LauraBRegional<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_LauraBRegional&d=DwMCaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=rDYMsRlLvZD78tOZEm6uEsr9BB-qbFg70Htrh99aTGQ&m=y5y8l0JnPGLkHEZXxJSeDxBc-gK-AxV7FT5e8q3bN6s&s=fuZhRJKxjlFnVgvj3o_xHLbx3tMoAaPEBbFuAT-aVMI&e=>, thanks for reporting. Could you please provide the content of the QGIS Help -> About menu ("click in the table, Ctrl+A and then Ctrl+C. Finally paste here.") as requested by the issue report form? Could you provide the GPKG layer or a sample GPKG layer with which the issue occurs?

**Comment by agiudiceandrea** [comment]
Hi @LauraBRegional, it seems to me the issue you are experiencing may have the same root cause of the one reported at https://github.com/qgis/QGIS/issues/53715.
If so, the issue has been already fixed with https://github.com/qgis/QGIS/pull/53756 in QGIS since version 3.28.9 and 3.32.1 and in GDAL with https://github.com/OSGeo/gdal/pull/8075 since version 3.7.2.
So, please try if the issue does occur using QGIS 3.28.9 or QGIS 3.32.1.

**Comment by LauraBRegional** [comment]
Hi Andrea,

We’ve tried with the new version 3.28.9 but we had the same result:
***@***.***



***@***.***


We lock the writing access to the files across Access Control List of Windows, I don't know if this information is useful for you... When the user isn’t locked there is any problem.

We have all the data on a set of Unix servers, the File Server is Unix, but it presents the data to us as if it were Windows. The most important thing is that the data is in a read-only shared folder, to avoid user errors.

Thanks!



                                  Laura Vergoñós
                                  Ambientòloga i analista SIG
                                  +34  93 223 82 98<tel:+34932238298>
[email-V3a-2.png]
P   Imprimeix només si és necessari

De: Andrea Giudiceandrea ***@***.***>
Enviado el: 26 July 2023 11:19
Para: qgis/QGIS ***@***.***>
CC: Laura Vergoñós Pascual ***@***.***>; Mention ***@***.***>
Asunto: Re: [EXTERN] [qgis/QGIS] Error reading layer in a GDB on a read-only drive QGIS 3.28.8 (Issue #53918)


Hi @LauraBRegional<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_LauraBRegional&d=DwMCaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=rDYMsRlLvZD78tOZEm6uEsr9BB-qbFg70Htrh99aTGQ&m=RfdIKup82K7losq3vs1-lZldZjG2lVrModNtdQZQWPI&s=1cRpWKZPHmWWZ9G0SpQZRKUa9n7674EWnV-uEhpOYmc&e=>, it seems to me the issue you are experiencing may have the same root cause of the one reported at #53715<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_qgis_QGIS_issues_53715&d=DwMCaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=rDYMsRlLvZD78tOZEm6uEsr9BB-qbFg70Htrh99aTGQ&m=RfdIKup82K7losq3vs1-lZldZjG2lVrModNtdQZQWPI&s=ZtxyrRxSHI0HNHwM95TlvcWKJvYiVLP3FgCZ0eVp2GI&e=>.
If so, the issue has been already fixed with #53756<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_qgis_QGIS_pull_53756&d=DwMCaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=rDYMsRlLvZD78tOZEm6uEsr9BB-qbFg70Htrh99aTGQ&m=RfdIKup82K7losq3vs1-lZldZjG2lVrModNtdQZQWPI&s=niDJ__WkHAB7FhoqYCe-bx_IrKXHR3_KoIuDnegRmOM&e=> in QGIS since version 3.28.9 and 3.32.1 and in GDAL with OSGeo/gdal#8075<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_OSGeo_gdal_pull_8075&d=DwMCaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=rDYMsRlLvZD78tOZEm6uEsr9BB-qbFg70Htrh99aTGQ&m=RfdIKup82K7losq3vs1-lZldZjG2lVrModNtdQZQWPI&s=eno3jbwoMhh-qoGWLKqhSvXe4ASzA9PIz7GytIuFoq8&e=> since version 3.7.2.
So, please try if the issue does occur using QGIS 3.28.9 or QGIS 3.32.1.

**Comment by agiudiceandrea** [comment]
Hi @rouault, have you seen this issue report about an issue that seems to me similar to (or the same as) https://github.com/qgis/QGIS/issues/53715 which was fixed by you in QGIS with https://github.com/qgis/QGIS/pull/53756?

**Comment by rouault** [comment]
@agiudiceandrea yes I've seen but I'm afraid I don't know how to address this without the exact setup to reproduce it

@LauraBRegional Can you open a OSGeo4W shell, run "python", and within it type the following (by replacing path\\to\\the.gdb with the actual path to the .gdb directory)
```
print(open("path\\to\\the.gdb\\a00000001.gdbtable", "r"))   # should succeed
print(open("path\\to\\the.gdb\\a00000001.gdbtable", "r+"))   # should throw an exception if this is really read-only
```

**Comment by agiudiceandrea** [comment]
Hi @LauraBRegional, does the issue still occur using the latest currently available QGIS versions (3.28.11 or 3.32.3)? If not, please provide the requested info:

> Can you open a OSGeo4W shell, run "python", and within it type the following (by replacing path\to\the.gdb with the actual path to the .gdb directory)
> 
> ```
> print(open("path\\to\\the.gdb\\a00000001.gdbtable", "r"))   # should succeed
> print(open("path\\to\\the.gdb\\a00000001.gdbtable", "r+"))   # should throw an exception if this is really read-only
> ```

**Comment by LauraBRegional** [comment]
Hi Andrea,

Sorry, the last month we’re on holidays and the answer was delayed of that.

Yes! Finally we realised the version 3.28.11 works fine and we don’t have the problem any more.

Thank you a lot for your support 😊




                                  Laura Vergoñós
                                  Ambientòloga i analista SIG
                                  +34  93 223 82 98<tel:+34932238298>
[email-V3a-2.png]
P   Imprimeix només si és necessari

De: Andrea Giudiceandrea ***@***.***>
Enviado el: 01 October 2023 08:29
Para: qgis/QGIS ***@***.***>
CC: Laura Vergoñós Pascual ***@***.***>; Mention ***@***.***>
Asunto: Re: [EXTERN] [qgis/QGIS] Error reading layer in a GDB on a read-only drive QGIS 3.28.8 (Issue #53918)


Hi @LauraBRegional<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_LauraBRegional&d=DwMCaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=rDYMsRlLvZD78tOZEm6uEsr9BB-qbFg70Htrh99aTGQ&m=TE9098uoVK7GdduyVD5Fcvs4sIv-0PxSXzIq3k_j70Mwq9xs2sIDzSPvYVYxqthz&s=4uN7z_jqQCBY7xXJ3zuNSPtNoCYAuZ1jH1jcwKb3Kms&e=>, does the issue still occur using the latest currently available QGIS versions (3.28.11 or 3.32.3)? If not, please provide the requested info:

Can you open a OSGeo4W shell, run "python", and within it type the following (by replacing path\to\the.gdb with the actual path to the .gdb directory)

print(open("path\\to\\the.gdb\\a00000001.gdbtable", "r"))   # should succeed

print(open("path\\to\\the.gdb\\a00000001.gdbtable", "r+"))   # should throw an exception if this is really read-only
